RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc

```diff
@@ -1023,6 +1034,70 @@ void MuonIdProducer::fillMuonId(edm::Event& iEvent,
     }
   }
 
+  // Fill GEM info
+  LogTrace("MuonIdentification") << "RecoMuon/MuonIdProducer :: fillMuonId :: fill GEM info";
+  if (gemHitHandle_.isValid()) {
+    for (const auto& chamber : info.chambers) {
+      // only GE1/1 and 2/1 are for rechits, reject station 0 and segments (layer==0 for GEMSegment)
+      if (chamber.id.subdetId() != MuonSubdetId::GEM || GEMDetId(chamber.id.rawId()).station() == 0 ||
+          GEMDetId(chamber.id.rawId()).layer() == 0)
+        continue;  // Consider RPC chambers only
+      const auto& lErr = chamber.tState.localError();
+      const auto& lPos = chamber.tState.localPosition();
+      const auto& lDir = chamber.tState.localDirection();
+
+      reco::MuonChamberMatch matchedChamber;
+
+      LocalError localError = lErr.positionError();
+      matchedChamber.x = lPos.x();
+      matchedChamber.y = lPos.y();
+      matchedChamber.xErr = sqrt(localError.xx());
+      matchedChamber.yErr = sqrt(localError.yy());
+
+      matchedChamber.dXdZ = lDir.z() != 0 ? lDir.x() / lDir.z() : 9999;
+      matchedChamber.dYdZ = lDir.z() != 0 ? lDir.y() / lDir.z() : 9999;
+      // DANGEROUS - compiler cannot guaranty parameters ordering
```

Comment:
Is it something that we need to monitor opening an issue?

Reply:
This was taken from the RPC code above, and ultimately looks to come from the original matching further up:
https://github.com/cms-sw/cmssw/blob/master/RecoMuon/MuonIdentification/plugins/MuonIdProducer.cc#L904

Reply:
Did you check that this is still a problem here? Why is it dangerous, who will have to deal with it?
I'm not sure if we should just be propagating warning comments in the code by copy-pasting code.
In fact, it might be useful to define a function rather than copy-past blocks of code...